olive/engine/engine.py

```diff
@@ -99,24 +103,24 @@ def __init__(
         self.pass_flows_search_spaces = None
 
         self.footprints = defaultdict(Footprint)
-        self.azureml_client_config = self._config.azureml_client_config
+        assert not kwargs
```

Comment:
since kwargs is not used and we don't allow them, isn't it equivalent to not having `**kwargs` in the init signature? 

Or is it added for future proofing?

Reply:
originally, I add it for check the additional unused parameters. I will remove it since passing redundant parameters and assert here has same result.